test/python/circuit/test_gate_definitions.py

```diff
@@ -113,7 +115,116 @@ def test_cx_definition(self):
+
+    @classmethod
+    def setUpClass(cls):
+
+        class_list = Gate.__subclasses__() + ControlledGate.__subclasses__()
+        exclude = {'ControlledGate', 'DiagonalGate', 'UCGate', 'MCGupDiag',
+                   'MCU1Gate', 'UnitaryGate', 'HamiltonianGate',
+                   'UCPauliRotGate', 'SingleQubitUnitary', 'MCXGate',
+                   'VariadicZeroParamGate'}
+        cls._gate_classes = []
+        for aclass in class_list:
+            if aclass.__name__ not in exclude:
+                cls._gate_classes.append(aclass)
+
+    def pgdef(self, gate):
+        for rule in gate.definition:
+            qubit_inds = [qubit.index for qubit in rule[1]]
+            params = np.array(rule[0].params)
+            print(f'{rule[0].name:5s} {str(qubit_inds):8s} {params} ')
+
+    def test_equivalence_phase(self):
+        """Test that the equivalent circuits from the equivalency_library
+        have equal matrix representations"""
+        for gate_class in self._gate_classes:
+        #for gate_class in [RYYGate]:
+            with self.subTest(i=gate_class):
+                n_params = len(_get_free_params(gate_class))
+                params = [0.1 * i for i in range(1, n_params+1)]
+                if gate_class.__name__ in ['MSGate']:
+                    params[0] = 2
+                elif gate_class in ['MCU1Gate']:
+                    params[1] = 2
+                gate = gate_class(*params)
+                equiv_lib_list = std_eqlib.get_entry(gate)
+                for ieq, equivalency in enumerate(equiv_lib_list):
+                    with self.subTest(msg=gate.name + '_' + str(ieq)):
+                        op1 = Operator(gate)
+                        op2 = Operator(equivalency)
+                        # print(gate.name)
+                        # np.set_printoptions(precision=4, linewidth=200, suppress=True)
+                        # print('')
+                        # print(gate.to_matrix_hide())
+                        # print('')                        
+                        # print(op1.data)
+                        # print('')                        
+                        # print(op2.data)
+                        # import ipdb;ipdb.set_trace()
+                        self.assertEqual(op1, op2)
+                        try:
+                            tomat = gate.to_matrix()
+                        except (AttributeError, QiskitError):
+                            pass
+                        else:
+                            self.assertTrue(np.allclose(op1.data, tomat))
+
+    def test_phase_gate(self):
+        from qiskit import transpile, QuantumRegister
+        np.set_printoptions(precision=2, linewidth=200, suppress=True)
+        def show_gates(gate):
+            print(gate.to_matrix())
+            for o, qreg, _ in gate.definition:
+                print(o.name, [qubit.index for qubit in qreg], o.params)
+                
+        class PhaseGate(Gate):
```

Comment:
What is this for? Is this just a `u1`? We should add a PhaseGate to Terra independently to replace U1 (see https://github.com/Qiskit/qiskit-terra/issues/4106), so I'm not sure if this is that (should go in the main code), or it's something else in which case it should change the name to not be confused with that.

Reply:
This was just temporary and has been removed.